### ISSUE TITLE
rename_identifiers not working

### ISSUE THREAD

**Issue by 3052** [wwhtrbbtt/deob-transformations#1]
~~~js
const transformations = require('deob-transformations');

const code = `
const c=n.a.get("API.lemonade.url"),d=n.a.get("API.lemonade.urlLinear"),u=n.a.get("API.lemonade.urlVod"),p=n.a.get("API.lemonade.platform"),m=n.a.get("API.lemonade.timeout");
`;

ast = transformations.code_to_ast(code);
transformations.rename_identifiers(ast);
deobbed = transformations.ast_to_code(ast);
console.log(transformations.beautify_code(deobbed));
~~~

result:

~~~js
const c = n.a.get("API.lemonade.url"),
  d = n.a.get("API.lemonade.urlLinear"),
  u = n.a.get("API.lemonade.urlVod"),
  p = n.a.get("API.lemonade.platform"),
  m = n.a.get("API.lemonade.timeout");
~~~

using another tool, I get:

~~~js
const const1_ = n.a.get("API.lemonade.url"),
  const2_ = n.a.get("API.lemonade.urlLinear"),
  const3_ = n.a.get("API.lemonade.urlVod"),
  const4_ = n.a.get("API.lemonade.platform"),
  const5_ = /* global */ n.a.get("API.lemonade.timeout");
~~~

https://richsnapp.com/tools/code-unmangler

using another tool, I get:

~~~js
const varFastenedWheel = n.a.get('API.lemonade.url'), varHairStepped = n.a.get('API.lemonade.urlLinear'), varFourRising = n.a.get('API.lemonade.urlVod'), varCourageYouth = n.a.get('API.lemonade.platform'), varBellRain = n.a.get('API.lemonade.timeout');
~~~

https://github.com/relative/synchrony

**Comment by pagpeter** [comment]
Hey,

The documentation is actually wrong here.
`rename_identifiers` only renames function arguments.
In many anti-bot systems, like Cloudflare, this has the same result, because all variables are function parameters ([example](https://cfschl.peet.ws/cdn-cgi/challenge-platform/h/b/scripts/jsd/61b90d1d/main.js))
I will update the documentation and the name, and will add another function to _actually_ rename identifiers

**Comment by 3052** [comment]
> I will update the documentation and the name, and will add another function to _actually_ rename identifiers

uh OK cool, but you closed the issue. are you gonna comment once that is done? or am I expected to just check the commits daily until its implemented?

**Comment by pagpeter** [comment]
Ill comment once I fixed it. 
You can create a PR too if you want / need it quickly, only a few lines.

**Comment by 3052** [comment]
I am using this for now:

https://richsnapp.com/tools/code-unmangler

it seems to be the best option of the ones I looked at - but if it gets added here I will check it out - thank you sir

**Comment by pagpeter** [comment]
I just released deob-transformations@1.1.0!

```js
const code = `const c=n.a.get("API.lemonade.url"),d=n.a.get("API.lemonade.urlLinear"),u=n.a.get("API.lemonade.urlVod"),p=n.a.get("API.lemonade.platform"),m=n.a.get("API.lemonade.timeout"); `;

const ast = transformations.code_to_ast(code);

transformations.rename_identifiers(ast);
transformations.rename_function_arguments(ast);

const deobbed = transformations.ast_to_code(ast);
console.log(transformations.beautify_code(deobbed));
```

Will now result in 

```js
const var_aaa = n.a.get("API.lemonade.url"),
  var_aab = n.a.get("API.lemonade.urlLinear"),
  var_aac = n.a.get("API.lemonade.urlVod"),
  var_aad = n.a.get("API.lemonade.platform"),
  var_aae = n.a.get("API.lemonade.timeout");
```
You can supply a custom set of names (like, commonly used, a list of first names for readability) by giving `rename_identifiers` a list of strings as the second argument. (E.g. `rename_identifiers(ast, ["John", "Jake", "Peet"])`)
Hope this covers everything you need.

An online solution, like code-unmangler, will come soon, probably [here](https://tools.peet.ws)

**Comment by 3052** [comment]
looks great, much appreciated
